src/Symfony/Component/Security/Core/Authorization/AuthorizationChecker.php

```diff
@@ -53,12 +54,10 @@ public function __construct(TokenStorageInterface $tokenStorage, AuthenticationM
             $this->tokenStorage->setToken($token = $this->authenticationManager->authenticate($token));
         }
 
-        if (!\is_array($attributes)) {
-            $attributes = [$attributes];
-        } else {
-            @trigger_error('Passing an array of Security attributes to '.__METHOD__.' is deprecated since Symfony 4.4. Use multiple isGranted() calls or the expression language (e.g. "has_role(...) or has_role(...)") instead.', \E_USER_DEPRECATED);
+        if (\is_array($attributes)) {
+            throw new InvalidArgumentException(sprintf('Passing an array of Security attributes to %s() is not supported.', __METHOD__));
         }
 
-        return $this->accessDecisionManager->decide($token, $attributes, $subject);
+        return $this->accessDecisionManager->decide($token, (array) $attributes, $subject);
```

Comment:
An array cast doesn't work here since `$attributes` can be an object, we could use `[$attributes]` instead since it can't be an array.

Reply:
only if $attributes is not null
would you mind sending a PR (with a test case ideally)?

Reply:
Even if `$attributes` is `null`, we should pass `[null]` to `decide`.

PR created : https://github.com/symfony/symfony/pull/33721